File: backend/src/main/java/com/paymybuddy/models/Transaction.java
package com.paymybuddy.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "transactions")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private User sender;

    @ManyToOne
    @JoinColumn(name = "receiver_id", nullable = false)
    private User receiver;

    @Column(name = "description", length = 255)
    private String description;

    @Column(name = "amount_in_cents", nullable = false)
    private Long amountInCents;

    @Column(name = "status", nullable = false)
    @Enumerated(EnumType.STRING)
    private TransactionStatus status = TransactionStatus.PENDING;
}

File: frontend/src/models/Transaction.ts
import { User } from './User'

export interface Transaction {
    id: number
    amountInCents: number
    description: string
    sender: User
    receiver: User
    createdAt: string
    status: string
}

export interface CreateTransactionRequest {
    receiverId: number
    amountInCents: number
    description: string
}

export function fromJson(json: any): Transaction {
    return {
        id: json.id,
        amountInCents: json.amountInCents,
        description: json.description ?? '',
        sender: json.sender,
        receiver: json.receiver,
        createdAt: json.createdAt,
        status: json.status,
    }
}
